utility/src/test/java/org/oppia/android/util/utility/NetworkConnectionTestUtilTest.kt

```diff
@@ -0,0 +1,159 @@
+package org.oppia.android.util.utility
+
+import android.app.Application
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkInfo
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.logging.EnableConsoleLog
+import org.oppia.android.util.logging.EnableFileLog
+import org.oppia.android.util.logging.GlobalLogLevel
+import org.oppia.android.util.logging.LogLevel
+import org.oppia.android.util.networking.NetworkConnectionUtil
+import org.oppia.android.util.networking.NetworkConnectionUtilProdModule
+import org.oppia.android.util.networking.ProdNetworkConnectionUtil
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Test for [NetworkConnectionTestUtil]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class NetworkConnectionTestUtilTest {
+
+  private val NO_CONNECTION = -1
+
+  @Inject
+  lateinit var context: Context
+
+  @Inject
+  lateinit var networkConnectionUtil: ProdNetworkConnectionUtil
```

Comment:
This seems a bit like a logical circular dependency to me. NetworkConnectionUtil is validated by using tests which depend on NetworkConnectionTestUtil. In order for NetworkConnectionTestUtil to be validated, it should be verified independently of NetworkConnectionUtil (otherwise it's kind of like trying to prove something with itself).

Could we just use the system service directly in the test to verify state? I suspect we don't need all the extra complexity of NetworkConnectionUtil, anyway.

Reply:
Can you please provide a reference on how can we do that?

Reply:
How would you verify this without NetworkConnectionUtilProdImpl? We can call directly into the system service like NetworkConnectionUtilProdImpl does to retrieve the state that we want to verify here.

Reply:
Updated the test file. PTAL. Thanks.